internal/goflags/flags.go

```diff
@@ -0,0 +1,169 @@
+// Package goflags allows parsing go command invocations and storing their flags in a
+// CommandFlags structure. It also provides utilities to backtrack through the process stack to
+// find and parse the flags of the first parent go command found in the process hierarchy.
+package goflags
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"strings"
+	"sync"
+
+	"github.com/datadog/orchestrion/internal/goproxy"
+	"github.com/shirou/gopsutil/v3/process"
+)
+
+// CommandFlags represents the flags provided to a go command invocation
+type CommandFlags struct {
+	Long    map[string]string
+	Short   []string
+	Unknown []string // flags we don't process but store anyway
+}
+
+var (
+	// specialFlags are long Go command flags that accept other flags as parameters
+	specialFlags = map[string]struct{}{
+		"-asmflags": {}, "-gccgoflags": {}, "-gcflags": {}, "-ldflags": {},
+	}
+	shortFlags = map[string]struct{}{
+		"-asan": {}, "-cover": {}, "-linkshared": {}, "-modcacherw": {}, "-msan": {}, "-race": {}, "-trimpath": {},
+	}
+	longFlags = map[string]struct{}{
+		"-covermode": {}, "-coverpkg": {}, "-buildmode": {}, "-buildvcs": {}, "-compiler": {}, "-mod": {},
+		"-modfile": {}, "-overlay": {}, "-pgo": {}, "-pkgdir": {}, "-tags": {},
+	}
```

Comment:
I wouldn't be afraid of using vertical space in there... Especially as this would make future updates more readable (also these should probably be sorted alphanumerically):

```suggestion
	specialFlags = map[string]struct{}{
		"-asmflags":   {}, // Flags passed through to the assembly
		"-gccgoflags": {}, // Flags passed through to the gccgo compiler
		"-gcflags":    {}, // Flags passed through to the gc compiler
		"-ldflags":    {}, // Flags passed through to the linker
	}
	shortFlags = map[string]struct{}{
		"-asan":       {}, // Enables address sanitizer
		"-cover":      {}, // Enables coverage colletion
		"-linkshared": {}, // Build code that links against shared libraries
		"-modcacherw": {}, // Keep module cache files read-write
		"-msan":       {}, // Enable memory sanitizer
		"-race":       {}, // Enable data race detection
		"-trimpath":   {}, // Remove all file system paths from the resulting executable
	}
	longFlags = map[string]struct{}{
		"-buildmode": {}, // Set build mode
		"-buildvcs":  {}, // Whether to stamp binaries with version control information
		"-compiler":  {}, // Select what compiler to use
		"-covermode": {}, // Set coverage mode
		"-coverpkg":  {}, // Set list of packages to collect coverage for
		"-mod":       {}, // Set module download mode
		"-modfile":   {}, // Set module file
		"-overlay":   {}, // Set overlay file
		"-pgo":       {}, // Set profile-guided optimization profile file
		"-pkgdir":    {}, // Set package install & load directory
		"-tags":      {}, // Set build tags
	}
```